std/stdio.d

```diff
@@ -725,7 +725,8 @@ $(D rawRead) always reads in binary mode on Windows.
     {
         import std.exception : enforce, errnoEnforce;
 
-        enforce(buffer.length, "rawRead must take a non-empty buffer");
+        if (!buffer.length)
+            enforce(false, "rawRead must take a non-empty buffer");
```

Comment:
This requires using `mixin` at call site. e.g.:

`mixin(enforce(cond, msg));`

which is ugly and awkward.

The pity here is that in 99% of cases, the message passed to enforce does not involve any computation. The whole point of using the lazy parameter is to defer construction of the exception message until you have confirmed the condition is false. If the construction takes 0 time, then there is no point to defer. An `inlineEnforce` function would be a kludge, but may get the job done for now.

Personally, I find `enforce` quite useless. It's not difficult to do `if(!cond) throw Exception(msg);`